src/Extension/SandboxExtension.php

```diff
@@ -50,16 +53,21 @@ public function disableSandbox()
         $this->sandboxed = false;
     }
 
-    public function isSandboxed()
+    public function isSandboxed($source = null)
     {
-        return $this->sandboxedGlobally || $this->sandboxed;
+        return $this->sandboxedGlobally || $this->sandboxed || $this->isSourceSandboxed($source);
     }
 
     public function isSandboxedGlobally()
     {
         return $this->sandboxedGlobally;
     }
 
+    public function isSourceSandboxed($source)
```

Comment:
is there any reason to make this method public ?

Reply:
No specific reason, just that all the rest of the functions in the class are public..

Reply:
That's because other methods are part of the public API.